Cubical/Algebra/IntegerMatrix/Diagonalization.agda

```diff
@@ -0,0 +1,283 @@
+{-
+
+This files contains a diagonalization precedure simpler than Smith normalization.
+But it is enough to establish important properties of finitely presented abelian groups.
+Also, it can be computed much more efficiently (than Smith, only).
```

Comment:
It would be good to specify what kind of diagonalization this produces

Reply:
done